src/services/codefixes/importFixes.ts

```diff
@@ -356,6 +358,25 @@ function createImportAdderWorker(sourceFile: SourceFile | FutureSourceFile, prog
         }
     }
 
+    function addImportForExternalModuleSymbol(symbol: Symbol, symbolName: string, isValidTypeOnlyUseSite: boolean, referenceImport?: ImportOrRequireAliasDeclaration) {
+        const useRequire = shouldUseRequire(sourceFile, program);
+        const moduleSpecifier = moduleSpecifiers.getLocalModuleSpecifierBetweenFileNames(
+            sourceFile,
+            Debug.checkDefined(symbol.valueDeclaration).getSourceFile().fileName,
+            compilerOptions,
+            createModuleSpecifierResolutionHost(program, host),
+        );
+        let info: FixInfo = {
+            fix: { kind: ImportFixKind.AddNew, importKind: ImportKind.Namespace, addAsTypeOnly: isValidTypeOnlyUseSite ? AddAsTypeOnly.Allowed : AddAsTypeOnly.NotAllowed, useRequire, moduleSpecifierKind: undefined, moduleSpecifier },
+            symbolName,
+            errorIdentifierText: undefined,
+        };
+        if (referenceImport && isImportClause(referenceImport)) {
+            info = { ...info, fix: { ...info.fix, importKind: ImportKind.Default, addAsTypeOnly: isTypeOnlyImportDeclaration(referenceImport) ? AddAsTypeOnly.Required : AddAsTypeOnly.NotAllowed } };

```

Comment:
I think `isValidTypeOnlyUseSite` should be consulted here in addition to the reference import. Also, what if the reference import is a type-only namespace import? It looks like we're only considering the type-onlyness of the import if it's a default import.

Reply:
Can you explain what the rules are supposed to be for setting `addAsTypeOnly`? I don't quite understand the last commit.

Reply:
I believe when `isValidTypeOnlyUseSite ` is true, then if `isTypeOnlyImportDeclaration(referenceImport)` is true then it should be `AddAsTypeOnly.Required` and if not, then it should be `AddAsTypeOnly.Allowed`.

Reply:
Yeah, and if `isValidTypeOnlyUseSite` is false, then it has to be `AddAsTypeOnly.NotAllowed`. That makes sense to me!